src/validators/SignedKeyRequestValidator.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+import {Ownable2Step} from "openzeppelin/contracts/access/Ownable2Step.sol";
+import {EIP712} from "openzeppelin/contracts/utils/cryptography/EIP712.sol";
+
+import {IMetadataValidator} from "../interfaces/IMetadataValidator.sol";
+import {IdRegistryLike} from "../interfaces/IdRegistryLike.sol";
+import {IdRegistry} from "../IdRegistry.sol";
+
+contract SignedKeyRequestValidator is IMetadataValidator, Ownable2Step, EIP712 {
+    /*//////////////////////////////////////////////////////////////
+                                 STRUCTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice Signed key request specific metadata.
+     *
+     *  @param requestingFid The fid of the entity requesting to add
+     *                       a signer key.
+     *  @param requestSigner Signer address. Must be the owner of
+     *                       the requestingFid fid.
+     *  @param signature     EIP-712 SignedKeyRequest signature.
+     *  @param deadline      block.timestamp after which signature expires.
+     */
+    struct SignedKeyRequest {
+        uint256 requestFid;
```

Comment:
nit: requestingFid

Reply:
I changed this away from the spec because I thought `requestFid` and `requestSigner` here were clearer and more consistent. `requestFid` is the fid of the signed key request, and `requestSigner` is the signer of the request. One field with an "-ing" and one without seemed inconsistent. Pretty minor, though, if you feel like it was clearer I'm not attached to the naming.